### DISCUSSION TITLE
Fedora 36 on Surface 3 keeps freezing constantly and randomly

### DISCUSSION THREAD

**Discussion by verduno** [linux-surface/linux-surface#852] [Q&A]
Dear All!

I wanted to rescue an old and not too powerful (2GB) Surface 3 (not pro), and installed Fedora 36 with the linux-surface kernel (latest 5.17.2). All the functions works like a charm, it is awesome, except that it freezes constantly and randomly. I have noticed that even though everything is frozen on the display or it turns black, when touching the keyboard, touchpad or the bluetooth mouse the keyboard backlight responses.
The last 2 journalctl logs are the following:

I would be incredibly happy if someone could give me advice. I have installed all the intel video codecs (RPM Fusion), and the acpitool. (I am unfortunately very beginner in linux.) Thanks in advance.

(The problem seems the same as here: https://ask.fedoraproject.org/t/fedora-keeps-freezing-constantly-and-randomly/18801/16)

Jun 12 01:49:56 fedora kernel: ACPI Error: Needed [Integer/String/Buffer], foun>
Jun 12 01:49:56 fedora kernel: ACPI Error: AE_TYPE, Opcode is not deferred [KEY>
Jun 12 01:49:56 fedora kernel: ACPI Error: AE_TYPE, Could not execute arguments>
Jun 12 01:49:56 fedora kernel: ACPI Error: AE_TYPE, Opcode is not deferred [KEY>
Jun 12 01:49:56 fedora kernel: ACPI Error: AE_TYPE, Could not execute arguments>
Jun 12 01:49:56 fedora kernel: ACPI Error: AE_TYPE, Opcode is not deferred [KEY>
Jun 12 01:49:56 fedora kernel: ACPI Error: AE_TYPE, Could not execute arguments>
Jun 12 01:49:56 fedora kernel: ACPI Error: AE_TYPE, Opcode is not deferred [KEY>
Jun 12 01:49:56 fedora kernel: ACPI Error: AE_TYPE, Could not execute arguments>
Jun 12 01:49:56 fedora kernel: ACPI Error: AE_TYPE, Opcode is not deferred [KEY>
Jun 12 01:49:56 fedora kernel: ACPI Error: AE_TYPE, Could not execute arguments>
Jun 12 01:49:56 fedora kernel: ACPI Error: Cannot assign type [Device] to [Buff>
Jun 12 01:49:56 fedora kernel: ACPI Error: Aborting method \_SB.PCI0.UOTG._CRS >
Jun 12 01:49:56 fedora kernel: ACPI Error: Method execution failed \_SB.PCI0.UO>
Jun 11 23:50:01 fedora systemd[1]: Failed to load BPF object: No such process
Jun 11 23:50:03 fedora systemd-udevd[519]: mmcblk1p2: /usr/lib/udev/rules.d/60->
Jun 11 23:50:03 fedora systemd-udevd[525]: mmcblk1p1: /usr/lib/udev/rules.d/60->
Jun 11 23:50:03 fedora systemd-udevd[521]: mmcblk1p3: /usr/lib/udev/rules.d/60->
Jun 11 23:50:03 fedora kernel: rt5640 i2c-10EC5640:00: Device with ID register >
Jun 11 23:50:03 fedora systemd-udevd[529]: mmcblk0p1: /usr/lib/udev/rules.d/60->
Jun 11 23:50:05 fedora /usr/sbin/irqbalance[658]: libcap-ng used by "/usr/sbin/>
Jun 11 23:50:05 fedora smartd[678]: In the system's table of devices NO devices>
Jun 11 23:50:05 fedora alsactl[741]: alsa-lib main.c:1541:(snd_use_case_mgr_ope>
lines 1-23...skipping...
Jun 12 01:49:56 fedora kernel: ACPI Error: Needed [Integer/String/Buffer], found [Device] 00000000f5994651 (20211217/exresop-388)
Jun 12 01:49:56 fedora kernel: ACPI Error: AE_TYPE, Opcode is not deferred [KEYS] (Region) (20211217/nsinit-411)
Jun 12 01:49:56 fedora kernel: ACPI Error: AE_TYPE, Could not execute arguments for [KEYS] (Region) (20211217/nsinit-419)
Jun 12 01:49:56 fedora kernel: ACPI Error: AE_TYPE, Opcode is not deferred [KEYS] (Region) (20211217/nsinit-411)
Jun 12 01:49:56 fedora kernel: ACPI Error: AE_TYPE, Could not execute arguments for [KEYS] (Region) (20211217/nsinit-419)
Jun 12 01:49:56 fedora kernel: ACPI Error: AE_TYPE, Opcode is not deferred [KEYS] (Region) (20211217/nsinit-411)
Jun 12 01:49:56 fedora kernel: ACPI Error: AE_TYPE, Could not execute arguments for [KEYS] (Region) (20211217/nsinit-419)
Jun 12 01:49:56 fedora kernel: ACPI Error: AE_TYPE, Opcode is not deferred [KEYS] (Region) (20211217/nsinit-411)
Jun 12 01:49:56 fedora kernel: ACPI Error: AE_TYPE, Could not execute arguments for [KEYS] (Region) (20211217/nsinit-419)
Jun 12 01:49:56 fedora kernel: ACPI Error: AE_TYPE, Opcode is not deferred [KEYS] (Region) (20211217/nsinit-411)
Jun 12 01:49:56 fedora kernel: ACPI Error: AE_TYPE, Could not execute arguments for [KEYS] (Region) (20211217/nsinit-419)
Jun 12 01:49:56 fedora kernel: ACPI Error: Cannot assign type [Device] to [BufferField] (must be type Int/Str/Buf) (20211217/exstoren-90)
Jun 12 01:49:56 fedora kernel: ACPI Error: Aborting method \_SB.PCI0.UOTG._CRS due to previous error (AE_AML_OPERAND_TYPE) (20211217/psparse-529)
Jun 12 01:49:56 fedora kernel: ACPI Error: Method execution failed \_SB.PCI0.UOTG._CRS due to previous error (AE_AML_OPERAND_TYPE) (20211217/uteval-68)
Jun 11 23:50:01 fedora systemd[1]: Failed to load BPF object: No such process
Jun 11 23:50:03 fedora systemd-udevd[519]: mmcblk1p2: /usr/lib/udev/rules.d/60-block-scheduler.rules:5 Failed to write ATTR{/sys/devices/pci0000:00/80860F14:00/mmc_host/mmc>
Jun 11 23:50:03 fedora systemd-udevd[525]: mmcblk1p1: /usr/lib/udev/rules.d/60-block-scheduler.rules:5 Failed to write ATTR{/sys/devices/pci0000:00/80860F14:00/mmc_host/mmc>
Jun 11 23:50:03 fedora systemd-udevd[521]: mmcblk1p3: /usr/lib/udev/rules.d/60-block-scheduler.rules:5 Failed to write ATTR{/sys/devices/pci0000:00/80860F14:00/mmc_host/mmc>
Jun 11 23:50:03 fedora kernel: rt5640 i2c-10EC5640:00: Device with ID register 0x6308 is not rt5640/39
Jun 11 23:50:03 fedora systemd-udevd[529]: mmcblk0p1: /usr/lib/udev/rules.d/60-block-scheduler.rules:5 Failed to write ATTR{/sys/devices/pci0000:00/80860F14:01/mmc_host/mmc>
Jun 11 23:50:05 fedora /usr/sbin/irqbalance[658]: libcap-ng used by "/usr/sbin/irqbalance" failed dropping bounding set due to not having CAP_SETPCAP in capng_apply
Jun 11 23:50:05 fedora smartd[678]: In the system's table of devices NO devices found to scan
Jun 11 23:50:05 fedora alsactl[741]: alsa-lib main.c:1541:(snd_use_case_mgr_open) error: failed to import hw:0 use case configuration -2
Jun 11 23:50:06 fedora kernel: Bluetooth: hci0: unexpected event for opcode 0x0000
Jun 11 23:50:06 fedora kernel: Bluetooth: hci0: unexpected event for opcode 0x0000
Jun 11 23:50:13 fedora dbus-broker-launch[980]: Ignoring duplicate name 'org.freedesktop.FileManager1' in service file '/usr/share//dbus-1/services/nemo.FileManager1.servic>
Jun 11 23:50:13 fedora gnome-session-binary[985]: GLib-GIO-CRITICAL: g_bus_get_sync: assertion 'error == NULL || *error == NULL' failed
Jun 11 23:50:13 fedora gnome-session-binary[985]: GLib-GIO-CRITICAL: g_bus_get_sync: assertion 'error == NULL || *error == NULL' failed
Jun 11 23:50:34 fedora gdm-password][1385]: gkr-pam: unable to locate daemon control file
Jun 11 23:50:37 fedora dbus-broker-launch[1517]: Ignoring duplicate name 'org.freedesktop.FileManager1' in service file '/usr/share//dbus-1/services/nemo.FileManager1.servi>
Jun 11 23:50:42 fedora kernel: i2c_designware 808622C1:04: controller timed out
Jun 11 23:50:42 fedora kernel: i915 0000:00:02.0: [drm] *ERROR* Failed to xfer payload of size (1) to reg (1)
Jun 11 23:50:43 fedora kernel: i2c_designware 808622C1:04: controller timed out
Jun 11 23:50:43 fedora kernel: i915 0000:00:02.0: [drm] *ERROR* Failed to xfer payload of size (1) to reg (22)
Jun 11 23:50:48 fedora gdm-launch-environment][944]: GLib-GObject: g_object_unref: assertion 'G_IS_OBJECT (object)' failed
Jun 11 23:51:31 fedora systemd[1]: Timed out waiting for device dev-ipts-15.device - /dev/ipts/15.

---

Jun 12 14:35:12 fedora kernel: ACPI Error: Needed [Integer/String/Buffer], found [Device] 000000008027ea79 (20211217/exresop-388)
Jun 12 14:35:12 fedora kernel: ACPI Error: AE_TYPE, Opcode is not deferred [KEYS] (Region) (20211217/nsinit-411)
Jun 12 14:35:12 fedora kernel: ACPI Error: AE_TYPE, Could not execute arguments for [KEYS] (Region) (20211217/nsinit-419)
Jun 12 14:35:12 fedora kernel: ACPI Error: AE_TYPE, Opcode is not deferred [KEYS] (Region) (20211217/nsinit-411)
Jun 12 14:35:12 fedora kernel: ACPI Error: AE_TYPE, Could not execute arguments for [KEYS] (Region) (20211217/nsinit-419)
Jun 12 14:35:12 fedora kernel: ACPI Error: AE_TYPE, Opcode is not deferred [KEYS] (Region) (20211217/nsinit-411)
Jun 12 14:35:12 fedora kernel: ACPI Error: AE_TYPE, Could not execute arguments for [KEYS] (Region) (20211217/nsinit-419)
Jun 12 14:35:12 fedora kernel: ACPI Error: AE_TYPE, Opcode is not deferred [KEYS] (Region) (20211217/nsinit-411)
Jun 12 14:35:12 fedora kernel: ACPI Error: AE_TYPE, Could not execute arguments for [KEYS] (Region) (20211217/nsinit-419)
Jun 12 14:35:12 fedora kernel: ACPI Error: AE_TYPE, Opcode is not deferred [KEYS] (Region) (20211217/nsinit-411)
Jun 12 14:35:12 fedora kernel: ACPI Error: AE_TYPE, Could not execute arguments for [KEYS] (Region) (20211217/nsinit-419)
Jun 12 14:35:12 fedora kernel: ACPI Error: Cannot assign type [Device] to [BufferField] (must be type Int/Str/Buf) (20211217/exstoren-90)
Jun 12 14:35:12 fedora kernel: ACPI Error: Aborting method \_SB.PCI0.UOTG._CRS due to previous error (AE_AML_OPERAND_TYPE) (20211217/psparse-529)
Jun 12 14:35:12 fedora kernel: ACPI Error: Method execution failed \_SB.PCI0.UOTG._CRS due to previous error (AE_AML_OPERAND_TYPE) (20211217/uteval-68)
Jun 12 12:35:18 fedora systemd[1]: Failed to load BPF object: No such process
Jun 12 12:35:19 fedora systemd-udevd[517]: mmcblk1p1: /usr/lib/udev/rules.d/60-block-scheduler.rules:5 Failed to write ATTR{/sys/devices/pci0000:00/80860F14:00/mmc_host/mmc>
Jun 12 12:35:19 fedora kernel: rt5640 i2c-10EC5640:00: Device with ID register 0x6308 is not rt5640/39
Jun 12 12:35:19 fedora systemd-udevd[527]: mmcblk1p2: /usr/lib/udev/rules.d/60-block-scheduler.rules:5 Failed to write ATTR{/sys/devices/pci0000:00/80860F14:00/mmc_host/mmc>
Jun 12 12:35:19 fedora systemd-udevd[522]: mmcblk1p3: /usr/lib/udev/rules.d/60-block-scheduler.rules:5 Failed to write ATTR{/sys/devices/pci0000:00/80860F14:00/mmc_host/mmc>
Jun 12 12:35:20 fedora systemd-udevd[519]: mmcblk0p1: /usr/lib/udev/rules.d/60-block-scheduler.rules:5 Failed to write ATTR{/sys/devices/pci0000:00/80860F14:01/mmc_host/mmc>
Jun 12 12:35:22 fedora /usr/sbin/irqbalance[661]: libcap-ng used by "/usr/sbin/irqbalance" failed dropping bounding set due to not having CAP_SETPCAP in capng_apply
Jun 12 12:35:22 fedora smartd[681]: In the system's table of devices NO devices found to scan
Jun 12 12:35:22 fedora alsactl[748]: alsa-lib main.c:1541:(snd_use_case_mgr_open) error: failed to import hw:0 use case configuration -2
Jun 12 12:35:23 fedora kernel: Bluetooth: hci0: unexpected event for opcode 0x0000
Jun 12 12:35:23 fedora kernel: Bluetooth: hci0: unexpected event for opcode 0x0000
Jun 12 12:35:30 fedora dbus-broker-launch[984]: Ignoring duplicate name 'org.freedesktop.FileManager1' in service file '/usr/share//dbus-1/services/nemo.FileManager1.servic>
Jun 12 12:35:30 fedora gnome-session-binary[989]: GLib-GIO-CRITICAL: g_bus_get_sync: assertion 'error == NULL || *error == NULL' failed
Jun 12 12:35:30 fedora gnome-session-binary[989]: GLib-GIO-CRITICAL: g_bus_get_sync: assertion 'error == NULL || *error == NULL' failed
Jun 12 12:36:17 fedora gdm-password][1393]: gkr-pam: unable to locate daemon control file
Jun 12 12:36:20 fedora dbus-broker-launch[1531]: Ignoring duplicate name 'org.freedesktop.FileManager1' in service file '/usr/share//dbus-1/services/nemo.FileManager1.servi>
Jun 12 12:36:25 fedora kernel: i2c_designware 808622C1:04: controller timed out
Jun 12 12:36:25 fedora kernel: i915 0000:00:02.0: [drm] *ERROR* Failed to xfer payload of size (1) to reg (1)
Jun 12 12:36:26 fedora kernel: i2c_designware 808622C1:04: controller timed out
Jun 12 12:36:26 fedora kernel: i915 0000:00:02.0: [drm] *ERROR* Failed to xfer payload of size (1) to reg (22)
Jun 12 12:36:30 fedora gdm-launch-environment][948]: GLib-GObject: g_object_unref: assertion 'G_IS_OBJECT (object)' failed
Jun 12 12:36:48 fedora systemd[1]: Timed out waiting for device dev-ipts-15.device - /dev/ipts/15.
Jun 12 12:42:33 fedora kernel: i2c_designware 808622C1:04: controller timed out
Jun 12 12:42:33 fedora kernel: i915 0000:00:02.0: [drm] *ERROR* Failed to xfer payload of size (1) to reg (1)
Jun 12 12:42:34 fedora kernel: i2c_designware 808622C1:04: controller timed out
Jun 12 12:42:34 fedora kernel: i915 0000:00:02.0: [drm] *ERROR* Failed to xfer payload of size (1) to reg (22)
Jun 12 12:42:45 fedora kernel: i915 0000:00:02.0: [drm] *ERROR* [CRTC:72:pipe B] flip_done timed out
lines 1-41

**Comment by NP-chaonay** [comment]
> keyboard backlight responses.

idk if this works at firmware or hardware level, maybe if there is linux api control that part of surface aggregator, then it might control. But on most system the surface kbd backlight is automatically controlled. (auto timeout is also auto controlling)

> ACPI Error: 
either 2 things: kernel bug or HW/firmware bug; but mostly kernel/driver bug

> Jun 11 23:50:13 fedora gnome-session-binary[985]: GLib-GIO-CRITICAL: g_bus_get_sync: assertion 'error == NULL || *error == NULL' failed
if I dont guess it wrong, at this moment user sesssion broken and maybe started GSOD (grey screen of death in GNOME)

if this just happens recently, the cause is not much clear
if this happens since install, then kernel/driver bug

**Reply by verduno** [reply]
Thanks for the reply.

It was working perfectly (but slowly of course) with Windows, so I think no HW issue.

YES it happens from the beginning, since first install after Win10 (grey screen of death in GNOME but it was the same with xorg).

Do you have any recommendation where to start to find out the problem?
